builtin/game/falling.lua

```diff
@@ -33,11 +33,105 @@ core.register_entity(":__builtin:falling_node", {
 				end
 			end
 		end
+		local def = core.registered_nodes[node.name]
+		if not def then
+			-- Don't allow unknown nodes to fall
+			minetest.log("warning", "Unknown falling node removed at "..minetest.pos_to_string(self.object:get_pos()))
+			self.object:remove()
+			return
+		end
 		self.meta = meta
-		self.object:set_properties({
-			is_visible = true,
-			textures = {node.name},
-		})
+		if def.drawtype == "airlike" then
+			self.object:set_properties({
+				is_visible = false,
+			})
+		else
+			if def.drawtype == "torchlike" or def.drawtype == "signlike" then
+				local textures
+				if def.tiles and def.tiles[1] then
+					if def.drawtype == "torchlike" then
+						textures = { "("..def.tiles[1]..")^[transformFX", def.tiles[1] }
+					else
+						textures = { def.tiles[1] }
+					end
+				end
+				self.object:set_properties({
+					is_visible = true,
+					visual = "upright_sprite",
+					visual_size = { x=1, y=1 },
+					textures = textures,
+					glow = def.light_source,
+				})
+			else
+				local itemstring = node.name
+				if core.is_colored_paramtype(def.paramtype2) then
+					itemstring = core.itemstring_with_palette(itemstring, node.param2)
+				end
+				self.object:set_properties({
+					is_visible = true,
+					wield_item = itemstring,
+					glow = def.light_source,
+				})
+			end
+		end
+		-- Rotate entity
+		if def.drawtype == "torchlike" then
+			self.object:set_yaw(math.pi*0.25)
+		elseif (node.param2 ~= 0 and (def.wield_image == "" or def.wield_image == nil)) or def.drawtype == "signlike" then
```

Comment:
115 characters (after tabulators). Please break it into two or three lines to keep it <= 80 characters.